update dbt_is_deleted to string (#6749)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/snapshots.md

| `dbt_valid_from` | The timestamp when this snapshot row was first inserted and became valid. | This column can be used to order the different "versions" of a record. | `snapshot_meta_column_names: {dbt_valid_from: start_date}` |
| `dbt_valid_to`   | The timestamp when this row became invalidated. For current records, this is `NULL` by default or the value specified in `dbt_valid_to_current`. | The most recent snapshot record will have `dbt_valid_to` set to `NULL` or the specified value.  | `snapshot_meta_column_names: {dbt_valid_to: end_date}` |
| `dbt_scd_id`     | A unique key generated for each snapshot row. | This is used internally by dbt. | `snapshot_meta_column_names: {dbt_scd_id: scd_id}` |
| `dbt_updated_at` | The `updated_at` timestamp of the source record when this snapshot row was inserted. | This is used internally by dbt. | `snapshot_meta_column_names: {dbt_updated_at: modified_date}` |
| `dbt_is_deleted` | A string value indicating if the record has been deleted. (`True` if deleted, `False` if not deleted). |Added when `hard_deletes='new_record'` is configured.  | `snapshot_meta_column_names: {dbt_is_deleted: is_deleted}` |

> All of these column names can be customized using the `snapshot_meta_column_names` config. Refer to the [Example](/reference/resource-configs/snapshot_meta_column_names#example) for more details.

Author: ```suggestion
All of these column names can be customized using the `snapshot_meta_column_names` config. Refer to this [example](/reference/resource-configs/snapshot_meta_column_names#example) for more details.
```

> The timestamps used for each column are subtly different depending on the strategy you use:

Author: ```suggestion
*The timestamps used for each column are subtly different depending on the strategy you use:
```

> *The timestamps used for each column are subtly different depending on the strategy you use:

Reviewer: I think we should leave the * out unless it's used as an endnote/footnote somewhere else for reference. We could use language we use elsewhere like **Note:** or similar

```suggestion
The timestamps used for each column are subtly different depending on the strategy you use:
```

Author: I thought the same but the asterisk is actually referenced in line 436, which is why I added it back it 😅

Reviewer: I didn't even see that. Pretty easy to miss both ways and might want to consider removing both but doesn't need to block this PR

- For the `timestamp` strategy, the configured `updated_at` column is used to populate the `dbt_valid_from`, `dbt_valid_to` and `dbt_updated_at` columns.

>   <Expandable alt_header="Details for the timestamp strategy">

Reviewer: Suggest a slight language change since the examples on not code execution examples (details) but rather the output. Maybe something like this:
```suggestion
  <Expandable alt_header="Sample results for the timestamp strategy">
```

  | -- | ------- | ---------------- | ---------------- | ---------------- | ---------------- |
  | 1  | pending | 2024-01-01 10:47 | 2024-01-01 10:47 | 2024-01-01 11:05 | 2024-01-01 10:47 |
  | 1  | shipped | 2024-01-01 11:05 | 2024-01-01 11:05 |                  | 2024-01-01 11:05 |

  Snapshot results with `hard_deletes='new_record'`:

  | id | status  | updated_at       | dbt_valid_from   | dbt_valid_to     | dbt_updated_at   | dbt_is_deleted |
  |----|---------|------------------|------------------|------------------|------------------|----------------|
  | 1  | pending | 2024-01-01 10:47 | 2024-01-01 10:47 | 2024-01-01 11:05 | 2024-01-01 10:47 | False          |
  | 1  | shipped | 2024-01-01 11:05 | 2024-01-01 11:05 | 2024-01-01 11:20 | 2024-01-01 11:05 | False          |
  | 1  | deleted | 2024-01-01 11:20 | 2024-01-01 11:20 |                  | 2024-01-01 11:20 | True           |

  </Expandable>

- For the `check` strategy, the current timestamp is used to populate each column. If configured, the `check` strategy uses the `updated_at` column instead, as with the timestamp strategy.

>   <Expandable alt_header="Details for the check strategy">

Reviewer: Same as above
```suggestion
  <Expandable alt_header="Sample results for the check strategy">
```
